app/models/piece.rb

```diff
@@ -56,37 +56,31 @@ def is_obstructed?(row_final, col_final)
 
     #horizonal case
     if row_init == row_final
-      return Piece.exists?(position_row: row_init, position_column: col_range)
+      return Piece.exists?(position_row: row_init, position_column: col_range, game_id: game_id)
 
       #vertical case
     elsif col_init == col_final 
-      return Piece.exists?(position_row: row_range, position_column: col_init)
+      return Piece.exists?(position_row: row_range, position_column: col_init, game_id: game_id)
 
       #diagonal case
-    elsif (row_final - row_init).abs == (col_final - col_init).abs
-      return Piece.exists?(position_row: row_range, position_column: col_range)
+    elsif ((row_final - row_init).to_f/(col_final - col_init).to_f ).abs == 1
+      return Piece.exists?(position_row: row_range, position_column: col_range, game_id: game_id)
     end
 
     #invalid input case
-    raise RuntimeError, "invalid input. Not diagnal, horizontal, or vertical."
+    raise RuntimeError, "invalid input. Not diagonal, horizontal, or vertical."
     
   end
 
   #Capture_Logic
   def move_to!(new_row, new_column)
     @pieces = Game.find(game_id).pieces
     @pieces.each do |piece|
-      if piece.position_row == new_row && piece.position_column == new_column
-        if piece.color == color
-          #We can't move to a place where our own pieces are!
-          raise RuntimeError, "You can't capture your own piece!"
-        else
-          #Setting to nil to indicate a piece has been captured
-          piece.position_row = nil
-          piece.position_column = nil
-        end
+      if piece.position_row == new_row && piece.position_column == new_column && piece.color != color
```

Comment:
Yeah, I realized this too. I tried making a flash message but it was not as straight forward as I thought it would be....will continue to update this!